h/static/scripts/annotator/plugin/pdf.coffee

```diff
@@ -1,302 +1,113 @@
-    index = parseInt div.parentNode.id.substr(13) - 1
-
-    # Look up the page
-    @pageInfo[index]
-
-  getDocumentFingerprint: -> @_app.documentFingerprint
-  getDocumentInfo: -> @_app.documentInfo
-
-# Annotator plugin for annotating documents handled by PDF.js
-class Annotator.Plugin.PDF extends Annotator.Plugin
-  pluginInit: ->
-    # We need dom-text-mapper
-    unless @annotator.plugins.DomTextMapper
-      console.warn "The PDF Annotator plugin requires the DomTextMapper plugin. Skipping."
-      return
-
-    @anchoring = @annotator.anchoring
-
-    @anchoring.documentAccessStrategies.unshift
-      # Strategy to handle PDF documents rendered by PDF.js
-      name: "PDF.js"
-      mapper: PDFTextMapper
-
-  # Are we looking at a PDF.js-rendered document?
-  _isPDF: -> PDFTextMapper.applicable()
-
-  # Extract the URL of the PDF file, maybe from the chrome-extension URL
-  _getDocumentURI: ->
-    uri = window.location.href
-
-    # We might have the URI embedded in a chrome-extension URI
-    matches = uri.match('chrome-extension://[a-z]{32}/(content/web/viewer.html\\?file=)?(.*)')
-
-    # Get the last match
-    match = matches?[matches.length - 1]
-
-    if match
-      decodeURIComponent match
-    else
-      uri
-
-  # Get a PDF fingerPrint-based URI
-  _getFingerPrintURI: ->
-    fingerprint = @anchoring.document.getDocumentFingerprint()
-
-    # This is an experimental URN,
-    # as per http://tools.ietf.org/html/rfc3406#section-3.0
-    "urn:x-pdf:" + fingerprint
-
-  # Public: get a canonical URI, if this is a PDF. (Null otherwise)
-  uri: ->
-    return null unless @_isPDF()
-
-    # For now, we return the fingerprint-based URI first,
-    # because it's probably more relevant.
-    # OTOH, we can't use it for clickable source links ...
-    # but the path is also included in the matadata,
-    # so anybody who _needs_ that can access it from there.
-    @_getFingerPrintURI()
-
-  # Try to extract the title; first from metadata, then HTML header
-  _getTitle: ->
-    title = @anchoring.document.getDocumentInfo().Title?.trim()
-    if title? and title isnt ""
-      title
-    else
-      $("head title").text().trim()
-
-  # Get metadata
-  _metadata: ->
-    metadata =
-      link: [{
-        href: @_getFingerPrintURI()
-      }]
-      title: @_getTitle()
-
-    documentURI = @_getDocumentURI()
-    if documentURI.toLowerCase().indexOf('file://') is 0
-        metadata.filename = new URL(documentURI).pathname.split('/').pop()
-    else
-        metadata.link.push {href: documentURI}
-
-    metadata
-
-  # Public: Get metadata (when the doc is loaded). Returns a promise.
-  getMetaData: =>
-    new Promise (resolve, reject) =>
-      if @anchoring.document.waitForInit?
-        @anchoring.document.waitForInit().then =>
-          try
-            resolve @_metadata()
-          catch error
-            reject "Internal error"
-      else
-        reject "Not a PDF dom mapper."
+    annotator.plugins.BucketBar?.update()
```

Comment:
Oh, but maybe I wanted the removed highlights to disappear from the bucket bar instantly.